### ISSUE TITLE
Disable refresh of Home > All Clusters on startup

### ISSUE THREAD

**Issue by visit1985** [headlamp-k8s/headlamp#2165]
I have a huge amount of clusters defined in `.kube/config` which are auto-detected on startup and refreshed by spawning background processes. Would be great if this could be disabled/optimized somehow.

![image](https://github.com/user-attachments/assets/022f4254-338a-4d59-8bff-621064a78b7a)

**Comment by joaquimrocha** [comment]
@illume , this is a good reason for your idea of optimizing the cluster connection in the home view.

**Comment by benlei** [comment]
Would also like this to be toggle-able as I have to log into multiple OIDC for various clusters, when all I want to do usually is just access 1 or a handful (as-needed).

**Comment by whitingj** [comment]
I just want to add onto this. We have a lot of clusters, all require OIDC, some require a vpn, and others can only be accessed by specific people. It is extremely disruptive to have headlamp attempt to connect to all the clusters. This causes a lot of logins, spawned processes (like in the original report) and just a lot of failures. It is very disruptive. The typical workflow is an engineer wants to get to one or two clusters and look at their workload.

**Comment by siegenthalerroger** [comment]
Hiya, just wondering what is blocking this feature from being developed? (I see that it is in the blocked stage of the roadmap)

**Comment by JeffMelton** [comment]
Yep, with ~5600 clusters in my kubeconfig, Headlamp is essentially unusable on startup.

**Comment by illume** [comment]
There's some discussion in https://github.com/kubernetes-sigs/headlamp/issues/2623 which was a duplicate issue.

**Comment by illume** [comment]
I had a number of conversations with people at Kubecon with this issue.

@joaquimrocha I think this issue should be prioritized. (I guess for v0.32.0?)

**Comment by joaquimrocha** [comment]
> I had a number of conversations with people at Kubecon with this issue.
> 
> [@joaquimrocha](https://github.com/joaquimrocha) I think this issue should be prioritized. (I guess for v0.32.0?)

Yeah. Let's finally take another look. Probably we can start looking into it again in maybe 1.5 months.

**Comment by siegenthalerroger** [comment]
not stale
